src/components/feedback/Callout.tsx

```diff
@@ -0,0 +1,124 @@
+import classnames from 'classnames';
+import type { JSX } from 'preact';
+
+import type { PresentationalProps } from '../../types';
+import type { IconComponent } from '../../types';
+import { downcastRef } from '../../util/typing';
+import { CautionIcon, CheckIcon, CancelIcon } from '../icons';
+
+type ComponentProps = {
+  icon?: IconComponent;
+  status?: 'notice' | 'error' | 'success';
+
+  size?: 'sm' | 'md' | 'lg' | 'custom';
+  variant?: 'outlined' | 'raised' | 'custom';
+  unstyled?: boolean;
+};
+
+type HTMLAttributes = Omit<JSX.HTMLAttributes<HTMLElement>, 'size' | 'icon'>;
+
+export type CalloutProps = PresentationalProps &
+  ComponentProps &
+  HTMLAttributes;
+
+/**
+ * Render a banner-like alert message with corresponding icon and coloring
+ */
+const Callout = function Callout({
+  children,
+  classes,
+  elementRef,
+
+  icon: Icon,
+  status = 'notice',
+
+  size = 'md',
+  variant = 'outlined',
+  unstyled = false,
+
+  ...htmlAttributes
+}: CalloutProps) {
+  const styled = !unstyled;
+  const themed = styled && variant !== 'custom';
+  const sized = styled && size !== 'custom';
+
+  let StatusIcon = Icon;
+  if (!StatusIcon) {
+    switch (status) {
+      case 'success':
+        StatusIcon = CheckIcon;
+        break;
+      case 'error':
+        StatusIcon = CancelIcon;
+        break;
+      /* istanbul ignore next */
+      default:
+      case 'notice':
```

Comment:
What's the reason to add `case 'notice'` if we are also capturing `default` case?